upstairs/src/volume.rs

```diff
@@ -610,6 +610,72 @@ impl BlockIO for Volume {
 
         Ok(())
     }
+    async fn activate_with_gen(&self, gen: u64) -> Result<(), CrucibleError> {
+        for sub_volume in &self.sub_volumes {
+            sub_volume.conditional_activate_with_gen(gen).await?;
```

Comment:
The correctness of this isn't obvious: why is it okay to consider yourself activated with a particular gen if a subvolume is activated with a _different_ gen?

Reply:
So, `activate_with_gen()` has been updated to:
1) Allow activation with a higher number than the upstairs was started with, but only if the upstairs is not active yet.
2) If we are active, and the generation number we are active with is the same as what we requested, then return OK,
3) Any other combination should cause `activate_with_gen()` to fail.

I'd like to eventually remove `activate_with_gen()`, but not here in this PR till as that will involve some more serious test restructuring, and I'm not ready for that yet.